juju-tutorials/juju-tricks.md

```diff
@@ -21,47 +21,51 @@ If you prefer chatting, visit us on [IRC](https://webchat.freenode.net/#juju).
+state=down; juju status --format oneline | grep "agent:${state}" | egrep -o '[a-z0-9-]+/[0-9]+' | sort -u | xargs -P4 -I@ bash -c 'unit=@; juju ssh $unit "sudo service jujud-unit-${unit/\//-} restart"'
 ```
 
-To restart machine in particular state (including LXCs) use the following command and adjust the first part, `state=<TARGET_STATE>`, to match state of the machines that you want to restart. The command, as it is, will restart machines in state `down`.
+To restart machine in a particular state (including LXDs) use the following command and adjust the first part, `state=<TARGET_STATE>`, to match the state of the machines that you want to restart. The command, as it is, will restart the machines in state `down`.
 
 ```console
 state=down; juju machines | grep -v 'Machine' | awk -v state=${state} '{if ($2 == state) print $1}' | xargs -P4 -I@ bash -c 'machine=@; svc=${machine//\//-}; svc=${svc%:}; juju ssh ${machine%:} "sudo service jujud-machine-${svc}" restart'
 ```
 
 ### Restarting all units and machines
 
-To restart all units, regardless of their state, run following command.
+To restart all units, regardless of their state, run the following command.
 
 ```console
-juju status --format oneline | xargs -l4 | column -t | egrep -o '[a-z0-9-]+/[0-9]+' | sort -u | xargs -P4 -I@ bash -c 'unit=@; juju ssh $unit "sudo service jujud-unit-${unit/\//-} restart"
+juju status --format oneline | egrep -o '[a-z0-9-]+/[0-9]+' | sort -u | xargs -P4 -I@ bash -c 'unit=@; juju ssh $unit "sudo service jujud-unit-${unit/\//-} restart"
 ```
 
-To restart all machines, regardless of their state, run following command.
+To restart all machines, regardless of their state, run the following command.
 
 ```console
 juju machines | grep -v 'Machine' | awk '{print $1}' | xargs -P4 -I@ bash -c 'machine=@; svc=${machine//\//-}; svc=${svc%:}; juju ssh ${machine%:} "sudo service jujud-machine-${svc}" restart'
 ```
 
+### Retrying failed hooks
+
+If there are multiple units with failed hooks, you can retry to re-run them with `juju resolved --all`
+
 ## Tricks on debugging unit relations
 
 Duration: N/A
 
 ### Inspecting relation data
 
-To see data of the specific relation on certain unit, run the following command and adjust variables `unit` and `relation`  to match unit and relation that you wish to inspect. This example command will print data from the relation `cni` of the unit `kubernetes-master/0`.
+You can use `juju show-unit` command to see details about the unit's relations, including relation data. For example, to inspect relation `cni` of the unit `kubernetes-master/0`.
 
 ```console
-unit=kubernetes-master/0;relation=cni; juju run --unit $unit "relation-ids $relation | xargs -I_@ sh -c 'relation-list -r _@ | xargs -I_U sh -c \"relation-get -r _@ - _U | sed s,^,_U:, 2>&1\"'"
+juju show-unit kubernetes-master/0 --endpoint cni
```

Comment:
Should an output example be included?

Reply:
I think that the output is pretty self-explanatory. Also, it can get pretty big, depending on the data in the unit, so I don't think it would add much value to include it as an example here.